src/control/createIndex.ts

```diff
@@ -54,3 +57,21 @@ export const createIndex = (api: IndexOperationsApi) => {
     }
   };
 };
+
+const waitUntilIndexIsReady = async (
+  api: IndexOperationsApi,
+  indexName: string,
+  seconds: number = 0
+) => {
+  try {
+    const indexDescription = await api.describeIndex({ indexName });
+    if (!indexDescription.status?.ready) {
+      await new Promise((r) => setTimeout(r, 1000));
+      return await waitUntilIndexIsReady(api, indexName, seconds + 1);
+    } else {
+      return `Index is ready after ${seconds} seconds`;
```

Comment:
**Note:** I'm a little unsure of how we should return out of here. Originally there were `console.log()` calls, but I wasn't sure if that was the best option either as a lot of our current calls return either undefined or the actual response entity. Definitely open to taking different approaches to both the successful return here and how we're throwing a possible error below.

Reply:
I think this should just `return;` undefined so that `createIndex` will have a consistent return type any time it is used (with or without the wait).

Debug output could be useful, though, definitely. We don't have a proper logging solution in place. For now you could just hack something with console log and an environment variable.

I might add a function like 

```
const debugLog = (str) {
      if (process && process.env && process.env.PINECONE_DEBUG) {
                 console.log(`Index ${indexName} is ready after ${seconds} seconds`;
              }
      }
}
```

It is important to check the `process` object is present before accessing it. Because people could use this in exotic contexts where that's not available.

Then I'd do something along the lines of
```
try {
    const indexDescription = await api.describeIndex({ indexName });
    if (!indexDescription.status?.ready) {
      await new Promise((r) => setTimeout(r, 1000));
      return await waitUntilIndexIsReady(api, indexName, seconds + 1);
    } else {
       debugLog(`Index ${indexName} is ready after ${seconds} seconds`)
       return;
    }
}
```

Reply:
Good call, I went ahead and added the `debugLog()` function to `src/utils/` with a small test file to validate the flag. If we want to expand this or make changes going forward I figured it'd be nice to situate it there for now.